webapp/tests/karma/unit/services/contact-view-model-generator.js

```diff
@@ -335,13 +360,24 @@ describe('ContactViewModelGenerator service', () => {
         chai.expect(model.reports[0].fields.patient_name).to.equal(childPerson.name);
       });
     });
+
+    it('adds heading to reports', () => {
+      const report = { _id: 'ab' };
+      const dataRecord = { _id: 'ab', validSubject: 'ac', subject: { value: 'ad' } };
+      stubSearch(null, [ report ]);
+      stubGetDataRecords(null, [ dataRecord ]);
+      return runReportsTest([], (model) => {
+        chai.expect(model.reports[0].heading).to.equal(dataRecord.subject.value);
+      });
+    });
```

Comment:
I think we should add a few tests here, eg:
- a data record without a valid subject
- a data record with no subject name
- no data record found
- an array of data records